fix: update sync plugin to work with the latest Electric sync server

samwillis opened the pull request:
This fix is something of a stop gap untill we make some larger changes to the plugin and bring in multi table/shape sync.

copiltembel commented:
@samwillis I've pushed a stylecheck fix but some tests are failing on CI due to `offset` property missing in Message.

samwillis commented:
@copiltembel :man_facepalming: I had fogged to run `pnpm typecheck` before pushing. Fixed now, just removed the offending props.

Ready for a final review.

copiltembel reviewed (approved):
LGTM!
